core/arch/aarchxx/mangle.c

```diff
@@ -3191,16 +3189,8 @@ create_ldax_from_stex(dcontext_t *dcontext, instr_t *strex,
 #ifdef AARCH64
     case OP_stlxp:
     case OP_stxp:
-        /* We treat A64 pair-4byte as single-8byte to handle ldxr;stxp. */
-        if (merge_pair) {
-            ASSERT(opsz == OPSZ_4);
-            *dest_reg = reg_resize_to_opsz(*dest_reg, OPSZ_8);
-            return INSTR_CREATE_ldaxr(dcontext, opnd_create_reg(*dest_reg), memop);
-        } else {
-            return INSTR_CREATE_ldaxp(
-                dcontext, regop, opnd_create_reg(reg_resize_to_opsz(dest_reg2, opsz)),
-                memop);
-        }
+        return INSTR_CREATE_ldaxp(
```

Comment:
Include a comment somewhere stating that we're aware we won't handle pair-singleton mismatches: make it clear that's an explicit design choice.

Reply:
I have changed to patch so that we do support these mismatches and added comments explaining how and why it is done that way.